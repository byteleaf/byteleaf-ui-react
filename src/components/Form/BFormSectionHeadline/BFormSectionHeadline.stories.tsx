import React from 'react';

import { BFormSectionHeadline } from './BFormSectionHeadline';
import { GrayBackground } from '../../../../.storybook/decorators';

export default {
  title: 'Form|BFormSectionHeadline',
};

export const BFormSectionHeadlineStory = () => (
  <BFormSectionHeadline>
    This is a Headline with a small padding at the top (grey background for demonstration)
  </BFormSectionHeadline>
);

BFormSectionHeadlineStory.story = {
  name: 'Default',
  decorators: [GrayBackground],
  parameters: {
    notes: 'A Headline component, used as a section title in forms.',
  },
};
