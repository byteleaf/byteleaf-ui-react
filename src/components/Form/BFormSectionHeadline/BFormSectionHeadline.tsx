import React from 'react';
import { styled, Typography } from '@material-ui/core';

const HeadlineWrapper = styled('div')(({ theme }) => ({
  paddingTop: theme.spacing(2),
}));

export const BFormSectionHeadline = ({ children }: { children: string }) => (
  <HeadlineWrapper>
    <Typography variant="h6">{children}</Typography>
  </HeadlineWrapper>
);
