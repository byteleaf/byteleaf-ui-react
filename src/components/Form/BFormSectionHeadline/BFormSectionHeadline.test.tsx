import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BFormSectionHeadline } from './BFormSectionHeadline';

describe('BFormSectionHeadline', () => {
  it('renders correctly', async () => {
    const { container } = render(<BFormSectionHeadline>Headline</BFormSectionHeadline>);

    expect(container).toMatchSnapshot();
  });
});
