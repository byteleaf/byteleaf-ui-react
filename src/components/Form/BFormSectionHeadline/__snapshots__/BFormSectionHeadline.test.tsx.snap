// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BFormSectionHeadline renders correctly 1`] = `
<div>
  <div
    class="div-root-1 div-root-2"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6"
    >
      Headline
    </h6>
  </div>
</div>
`;
